src/rust/jsg/v8.rs

```diff
@@ -462,3 +481,67 @@ impl<'a> FunctionCallbackInfo<'a> {
         }
     }
 }
+
+/// A safe wrapper around a V8 isolate pointer.
+///
+/// `Isolate` provides a type-safe abstraction over raw `v8::Isolate*` pointers,
+/// ensuring that the pointer is always non-null. This type is `Copy` and can be
+/// freely passed around without worrying about ownership.
+///
+/// # Thread Safety
+///
+/// V8 isolates are single-threaded. While `Isolate` itself is `Send` and `Sync`
+/// (as it's just a pointer wrapper), V8 operations must only be performed on the
+/// thread that owns the isolate lock. Use `is_locked()` to verify the current
+/// thread holds the lock before performing V8 operations.
+///
+/// # Example
+///
+/// ```ignore
+/// // Create from raw pointer (unsafe)
+/// let isolate = unsafe { v8::Isolate::from_raw(raw_ptr) };
+///
+/// // Check if locked before V8 operations
+/// assert!(isolate.is_locked());
+///
+/// // Get raw pointer for FFI calls
+/// let ptr = isolate.as_ptr();
+/// ```
+#[derive(Clone, Copy, Debug)]
+pub struct Isolate {
```

Comment:
I would still call it IsolatePtr. Having value semantics for something that is called Isolate is weird.